### PR TITLE
failing if tar upload is in progress

### PR DESCRIPTION
**Pull request by antialias**

### CODE REVIEW

**Comment by antialias** [comment]
This should fail the tar upload if an identical upload is in progress. Can you test it?

**Comment by cheryly279** [comment]
@antialias It failed but not exactly how I expected. When I tried to do two at the same time they both failed, with a not entirely clear message (no such file or directory). I included screenshots. Once both failed I tried again from just one place and that worked successfully.

Are we expecting them both to fail? I would have thought one would pass.
screen shot 2016-03-14 at 11 00 44 am
screen shot 2016-03-14 at 10 58 32 am

**Comment by antialias** [comment]
Looks like the command failed when running cleaninstall. What happens when
you try to run cleaninstall by itself?

File: src/npm-cache-modules.sh [hunk 1]
```diff
@@ -50,6 +50,11 @@ echo "Hash for your package.json is $modulesHash"
 echo "Creating tgz of your node_modules in $tarPath. This may take a while if node_nodules is big."
 $tar czf $tarPath node_modules
 $ssh $host "mkdir -p $hostDest"
+if ! $ssh $host stat $hostTarPath
```
Comment (cheryly279) [inline src/npm-cache-modules.sh]: Same issue as before, when I take out the ! I expect to see the error message, but I don't. And I definitely made sure this time that my install of the code had my change. Is that how this works, anyway? When would stat return something that would be considered false?

Comment (antialias) [inline src/npm-cache-modules.sh]: the stat command should exit with non-zero if there is nothing at the path it was given.

**Comment by cheryly279** [comment]
Works now! Thx.

**Pull request merged**